cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -463,6 +463,42 @@ def test_snapshot_history_btn(create_vm, provider):
     assert snapshot_view.is_displayed
 
 
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1395116])
+def test_snapshot_link_after_delete(create_vm, provider):
+    """Tests snapshot link and history button after delete
+    Metadata:
+        test_flag: snapshot
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: medium
+        initialEstimate: 1/6h
+    Bugzilla:
+        1395116
+    """
+    has_name = not provider.one_of(RHEVMProvider)
+    snapshot1 = new_snapshot(create_vm, has_name=has_name)
+    snapshot1.create()
+    snapshot2 = new_snapshot(create_vm, has_name=has_name)
+    snapshot2.create()
+    snapshot1.delete()
```

Comment:
This looks like it is part of the setup for the asserts below. Thoughts on moving it into a fixture?

Reply:
Can you elaborate on why we'd use a fixture here? It would only be used by this test unless I updates a bunch of tests. And this pattern is used in several tests. Are you suggesting I use a fixture that will return 2 snapshots? I'm just not sure what value a fixture would give here, but open to learn.

Reply:
So in my mind, test setup should usually be part of a fixture, so that it doesn't impact the actual assertion of the tests. The purpose of this test is to check that those buttons are properly displayed _after_ a snapshot delete. Therefore, creating those snapshots is part of the setup for the test. 

We definitely don't follow this rule everywhere. And your point about the fixture not being used anywhere else is a good one. I do still think it is good practice to have test setup exist outside the main test. If this were to be a fixture, it wouldn't have to yield anything, i.e. it could just be

```python
@pytest.fixture
def create_snaps_and_delete():
    # ... create snaps
    # ... delete a snap
    yield
    # ... do cleanup (if necessary)
```
Overall though, I'd consider my comment _optional_.

Reply:
That sounds fair. Thanks for taking the time to look at my PRs. I'll put the snapshot creation and deletion into a fixture.